src/cli/commands/add.js

```diff
@@ -155,9 +155,27 @@ export class Add extends Install {
       reqDepth: 0,
     };
     const {trees, count} = await buildTree(this.resolver, this.linker, patterns, opts, true, true);
-    this.reporter.success(
-      count === 1 ? this.reporter.lang('savedNewDependency') : this.reporter.lang('savedNewDependencies', count),
-    );
+
+    if (count === 1) {
+      this.reporter.success(this.reporter.lang('savedNewDependency'));
+    } else {
+      this.reporter.success(this.reporter.lang('savedNewDependencies', count));
+    }
+
+    if (!count) {
+      return;
+    }
+
+    const resolverPatterns = new Set();
+    for (const pattern of patterns) {
+      const {version, name} = this.resolver.getResolvedPattern(pattern) || {};
+      resolverPatterns.add(`${name}@${version}`);
+    }
+    const directRequireDependencies = trees.filter(({name}) => resolverPatterns.has(name));
+
+    this.reporter.info(this.reporter.lang('directDependencies'));
+    this.reporter.tree('newDependencies', directRequireDependencies);
```

Comment:
I think we might want different keys instead of reusing `'newDependencies'` for the 2 calls to `.tree(`.  It doesn't matter for the console-reporter, but if you `--json` to use the json-reporter, this key could be used to differentiate the two trees if parsing the json output. Maybe change this one to `'newDirectDependencies'`.

Reply:
ah, that's what it was used for! Will fix :)